sphinxext/rediraffe.py

```diff
@@ -320,19 +319,57 @@ def abs_path_in_src_dir_w_src_suffix(filename: str) -> Union[Path, None]:
                 logger.error(err_msg)
                 self.app.statuscode = 1
 
+        file = open(path, "a")
+
         for renamed_file in rename_hints:
             hint_to, perc = rename_hints[renamed_file]
             if renamed_file in absolute_redirects:
                 logger.info(
                     f"renamed file {renamed_file} redirects to {absolute_redirects[renamed_file]}."
                 )
             else:
-                err_msg = (
-                    f"{red('(broken)')} {renamed_file} was deleted but is not redirected!"
-                    f" Hint: This file was renamed to {hint_to} with a similarity of {perc}%."
-                )
-                logger.error(err_msg)
-                self.app.statuscode = 1
+                if self.app.config.rediraffe_auto_redirect and not path.is_file():
+                    logger.warning(
+                        f"{red('(broken)')} Automatic redirects is only available with a redirects file."
+                    )
+                elif (
+                    self.app.config.rediraffe_auto_redirect
+                    and "rediraffewritediff" in self.name
+                ):
+                    if path_rename_from not in absolute_redirects:
+                        if perc >= self.app.config.rediraffe_auto_redirect_perc:
+                            from_fancy_path = str(renamed_file)[
+                                len(self.app.srcdir) + 1 :
+                            ]
+
+                            from_fancy_path = (
+                                '"' + from_fancy_path.replace("\\", "/") + '"'
+                            )
+
+                            to_fancy_path = str(hint_to)[len(self.app.srcdir) + 1 :]
+
+                            to_fancy_path = '"' + to_fancy_path.replace("\\", "/") + '"'
+
+                            file.write(from_fancy_path + " " + to_fancy_path + "\n")
+                            logger.info(
+                                f"{green('(okay)')} Renamed file {renamed_file} has been redirected to {hint_to} in your redirects file!"
+                            )
+                        else:
+                            err_msg = (
+                                f"{red('(broken)')} {renamed_file} was deleted but is not redirected!"
+                                f" Hint: This file was renamed to {hint_to} with a similarity of {perc}%."
+                            )
+                            logger.error(err_msg)
+                            self.app.statuscode = 1
+                else:
+                    err_msg = (
+                        f"{red('(broken)')} {renamed_file} was deleted but is not redirected!"
+                        f" Hint: This file was renamed to {hint_to} with a similarity of {perc}%."
+                    )
+                    logger.error(err_msg)
+                    self.app.statuscode = 1
```

Comment:
ew consecutive code duplication. I'm wanna play around with this section (324-370). It's getting really verbose.